modules/carto/src/layers/post-process-layer.ts

```diff
@@ -0,0 +1,183 @@
+import {Framebuffer, TextureProps} from '@luma.gl/core';
+import {
+  CompositeLayer,
+  Layer,
+  LayerContext,
+  LayersList,
+  PostProcessEffect,
+  PostRenderOptions
+} from '@deck.gl/core';
+
+const TEXTURE_PROPS: TextureProps = {
+  format: 'rgba8unorm',
+  mipmaps: false,
+  sampler: {
+    minFilter: 'linear',
+    magFilter: 'linear',
+    addressModeU: 'clamp-to-edge',
+    addressModeV: 'clamp-to-edge'
+  }
+};
+
+interface IPostProcessLayer {
+  applyPostProcess: () => void;
+  enableOffscreen: Layer['draw'];
+  disableOffscreen: () => void;
+}
+
+type Constructor<T> = (new (...args: any[]) => T) & {layerName: string};
+// type DrawableLayer = Layer & {
+//   initializeState(context: LayerContext): void;
+// };
+type DrawableCompositeLayer = CompositeLayer & {
+  renderLayers(): Layer<{}> | null | LayersList;
+};
+
+/**
+ * Dummy Layer that draws nothing, just calls back to root Layer
+ */
+class DrawCallbackLayer extends Layer {
+  static layerName = 'DrawCallbackLayer';
+
+  initializeState(this: DrawCallbackLayer): void {
+    this.id = `draw-callback-${this.root.props.id}`;
+  }
+
+  _drawLayer(this: DrawCallbackLayer) {
+    (this.root as unknown as IPostProcessLayer).applyPostProcess();
+  }
+}
+
+// export function OffscreenModifier<T extends Constructor<DrawableLayer>>(BaseLayer: T) {
+export function OffscreenModifier(BaseLayer) {
+  return class OffscreenLayer extends BaseLayer {
```

Comment:
Nit – for me the term "offscreen" implies use of OffscreenCanvas and Web Workers. Would one of these be clearer?

- RenderToTextureModifier
- RTTModifier
- RenderTargetModifier

Reply:
I've gone with `RTT` as it can then be used in other places, e.g. `enableRTT()`